src/services/userAuth.js

```diff
@@ -0,0 +1,97 @@
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+import uuid from 'uuid/v1';
+import dotenv from 'dotenv';
+
+import database from '../models/index';
+import doesUserexist from '../helpers/userHelper';
+import notification from './notification';
+
+dotenv.config();
+
+const saltRounds = 10;
+const users = database.User;
+
+/**
+ * Signs in a user
+ * @param {Object} request - The request object to the server
+ * @param {Object} response - The response object from the server
+ * @param {Function} Next - Calls the express error handler
+ * @return {Void} - Returns nothing
+ */
+export const createUser = async (request, response, Next) => {
+  const userPresent = await doesUserexist(request.body.email);
+  if (!userPresent) {
+    const hash = await bcrypt.hashSync(request.body.password, saltRounds);
+    const userData = { ...request.body, password: hash, id: uuid() };
+
+    const createdUser = await users.create(userData);
+    const { password, ...newUser } = createdUser.dataValues;
+    const host = request.get('host');
+    notification.emit('notification', {
+      type: 'accountActivation',
+      payload: [{
+        email: newUser.email,
+        name: newUser.firstName,
+        verificationLink: `https://${host}/profile`
+      }]
+    });
+
+    const token = jwt.sign(
+      { id: createdUser.dataValues.id },
+      process.env.SECRET_KEY,
+      { expiresIn: '14d' }
+    );
+
+    response.status(200).json({ status: 200, user: newUser, token });
+  } else {
+    const err = new Error();
+    err.message = 'you are already registered';
+    err.status = 409;
+    Next(err);
+  }
+};
+
+/**
+ * Creates a user
+ * @param {Object} request - The request object to the server
+ * @param {Object} response - The response object from the server
+ * @param {Function} Next - Calls the express error handler
+ * @return {Void} - Returns nothing
+ */
+export const checkinUser = async (request, response, Next) => {
+  const userData = { ...request.body };
+  const createdUser = await users.findAll({
+    where: {
+      email: userData.email
+    }
+  });
+
+  if (createdUser.length === 0) {
+    const err = new Error();
+    err.message = 'you are not yet registered';
+    err.status = 400;
+    Next(err);
+  } else {
+    const passwordTrue = await bcrypt.compareSync(
+      userData.password,
+      createdUser[0].dataValues.password
+    );
+
+    if (passwordTrue) {
+      const { password, ...user } = createdUser[0].dataValues;
+      const token = jwt.sign(
+        { id: user.id },
+        process.env.SECRET_KEY,
+        { expiresIn: '14d' }
```

Comment:
Please can you set the `expiring time` to `24hr` or less

Reply:
Why 24 hours?
I believe I asked this question on the group and you did not respond.
@iverenshaguy  suggested 2 weeks and I am going with that.

Reply:
for now, it should remain as 2 weeks till we have genuine reasons to change it

Reply:
What did the team agree on?

Reply:
We agreed on 24hrs

Reply:
Any particular reason(s)?

Reply:
Since we will be blacklisting tokens, we do not want tokens to have long TTL

Reply:
Makes sense. @successgilli Kindly fix

Reply:
okay